lib/core/network/router.js

```diff
@@ -138,6 +138,33 @@ class Router {
         this._executeFromHttp(route.verb, request, cb);
       });
     }
+
+    /**
+     * Returns inner metrics from the router
+     * @returns {Object}
+     */
+    global.kuzzle.onAsk(
+      'core:network:router:metrics',
+      () => this.metrics());
+  }
+
+  /**
+   * Returns the metrics of the router
+   * @returns {Object}
+   */
+  metrics () {
+    const connectionsByProtocol = new Map();
```

Comment:
Why using a map here? Usually we use Map for RAM cache or anything that need fast set/get. Here you use it as an intermediary storage object. Also you have to convert it to a POJO at the end of this method